tagstudio/src/core/library.py

```diff
@@ -2315,3 +2175,340 @@ def sort_fields(self, entry_id: int, order: list[int]) -> None:
+                return list(pre_results[0])
+            elif len(pre_results) == 0:
+                return []
+            result_set = set(pre_results[0])
+            # entries should be found in every part of query
+            for and_result in pre_results[1:]:
+                set_copy: list = list(result_set)
+                for recorded_entry in set_copy:
+                    entry_detected: bool = False
+                    for new_entry in and_result:
+                        if recorded_entry == new_entry:
+                            entry_detected = True
+                            break
+                    if not entry_detected:
+                        result_set.remove(recorded_entry)
+            return list(result_set)
+        # Entries should match any of parts separated by '|'
+        elif search_mode == SearchMode.OR:
+            for or_result in pre_results:
+                for or_entry in or_result:
+                    result_set.add(or_entry)
+            return list(result_set)
+
+    def handle_common_field(
+        self, entry_value: list[str] | str | None, value: str, query_is_negative: bool
+    ) -> bool:
+        if isinstance(entry_value, list):
+            entry_value = "".join(entry_value)
+
+        if entry_value is None:
+            return query_is_negative
+        else:
+            if value in entry_value:
+                return not query_is_negative
+            else:
+                return query_is_negative
+
+    def handle_tag_id(self, query: str, entry_tags: list[int]) -> bool:
+        id_query: int = 0
+        try:
+            id_query = int(query)
+        except Exception:
+            return False
+        return id_query in entry_tags
+
+    def handle_unbound(
+        self,
+        entry: Entry,
+        all_tag_terms: list[str],
+        entry_tags: list[int],
+        entry_authors: list[list[str]],
+        unbound: list[str],
+    ) -> tuple[ItemType, int] | None:
+        collations_added: list = []
+
+        special_flags = SpecialFlag("".join(unbound))
+
+        if self.add_entries_from_special_flags(
+            special_flags, entry_tags, entry_authors, entry
+        ):
+            return (ItemType.ENTRY, entry.id)
+
+        def add_entry(entry: Entry) -> tuple[ItemType, int] | None:
+            # self.filter_entries.append()
+            # self.filtered_file_list.append(file)
+            # results.append((SearchItemType.ENTRY, entry.id))
+            added = False
+            for f in entry.fields:
+                if self.get_field_attr(f, "type") == "collation":
+                    if self.get_field_attr(f, "content") not in collations_added:
+                        collations_added.append(self.get_field_attr(f, "content"))
+                        return (
+                            ItemType.COLLATION,
+                            self.get_field_attr(f, "content"),
+                        )
+                    added = True
```

Comment:
afaik `collation` isnt currently implemented, so this doesnt do anything at this moment, and `added` will be always `False`, right?

Reply:
@yedpodtrzitko 
This part is basically copy-paste from original code. I left it for backward compatability, but it is safe to remove, if feature was never implemented